### PR TITLE
chore(docs): complete Epic 32.1 fix feasibility spike

### PR DESCRIPTION
**Pull request by Alberto-Codes**
This PR delivers the Story 32.1 fix feasibility spike — a research spike validating that AST-based docstring insertion works reliably for the upcoming `docvet fix` command (Epic 32). Previously, there was no validated approach for inserting scaffolded docstring sections, and the scaffold finding category had no design spec.

- Evaluate 3 insertion strategies (string ops + AST, textwrap/inspect, libcst) and select string ops + AST line numbers (zero runtime deps)
- Build and verify proof-of-concept across 14 test scenarios including edge cases (nested classes, decorators, async, raw docstrings, tabs)
- Design scaffold finding category spec: `scaffold` category on Finding, `[TODO: describe]` placeholder, `scaffold-incomplete` rule, exit code and reporting integration
- Produce GO recommendation for Story 32.2 with risk assessment and edge case catalog
- Add planning artifacts: Epic 32 sprint change proposal, epics update, tech specs from refactoring extractions, docvet pitch page

chore(docs): add tech specs from refactoring extractions
chore(docs): add Epic 32 sprint change proposal and epics update
chore(docs): move docvet pitch page to planning artifacts

Test: CI only — no production code changed, all files are `_bmad-output/` artifacts

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
Decision document quality (32-1-spike-decision.md): insertion strategy rationale, scaffold category design, and the config pattern divergence note in section 4.4 (M4 review finding — Story 32.2 must resolve whether scaffold fail-on uses per-category boolean or extends existing `fail-on` list pattern).

### Related
- Story 32.2 (fix core section scaffolding engine) — builds on this spike's GO recommendation
- Epic 32: Linter Lifecycle — Fix & Suppress

### CODE REVIEW

File: _bmad-output/implementation-artifacts/32-1-spike-decision.md [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    line: int
+    symbol: str
+    rule: str
+    message: str
+    category: Literal["required", "recommended", "scaffold"]
+```
+
+**Severity**: `scaffold` findings are **required** severity — they indicate sections that exist structurally but have placeholder content that must be filled in. The `docvet fix` command inserts the scaffolding; scaffold findings enforce that users complete the content.
+
+**Rationale for required severity**: The purpose of `docvet fix` is to reduce friction, not to silence findings. A scaffolded `Raises: [TODO: describe]` is better than nothing (the section header exists, so the enrichment rule is satisfied), but the placeholder content is not documentation. Scaffold findings ensure the user follows through.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply _bmad-output/implementation-artifacts/32-1-spike-decision.md]: Addressed in latest push — rewording adopted. Now reads "category = scaffold, blocking by default" instead of mixing category and severity terms.

File: _bmad-output/implementation-artifacts/tech-spec-enrichment-forward-extraction.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
+---
+title: 'Extract forward checks into enrichment sub-package'
+slug: 'enrichment-forward-extraction'
+created: '2026-03-22'
+status: 'ready-for-dev'
+stepsCompleted: [1, 2, 3, 4]
+tech_stack: ['Python 3.12+']
+files_to_modify:
+  - src/docvet/checks/enrichment.py (rename to enrichment/__init__.py)
+  - src/docvet/checks/enrichment/_forward.py (new, ~658 lines)
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply _bmad-output/implementation-artifacts/tech-spec-enrichment-forward-extraction.md]: Addressed in latest push — updated status to `done`.
